lnx-engine/search-index/src/schema.rs

```diff
@@ -0,0 +1,538 @@
+use std::iter::FromIterator;
+
+use anyhow::{anyhow, Error, Result};
+use hashbrown::{HashMap, HashSet};
+use serde::{Deserialize, Serialize};
+use tantivy::schema::{
+    Cardinality,
+    FacetOptions,
+    Field,
+    FieldType,
+    IndexRecordOption,
+    IntOptions,
+    Schema,
+    SchemaBuilder,
+    TextFieldIndexing,
+    TextOptions,
+    FAST,
+    INDEXED,
+    STORED,
+};
+use tantivy::Score;
+
+use crate::helpers::{Calculated, Validate};
+
+pub static PRIMARY_KEY: &str = "_id";
+
+fn default_to_true() -> bool {
+    true
```

Comment:
I'm curious what the purpose of this is.

Reply:
This can be ignored for now, alot of this is WIP and has a load of code warnings left in it while debugging unittest issues.

Will be removed later 👌

Reply:
This is a rather unfortunate result of serde not allowing default values directly and instead requiring a factory.

This is basically doing "if it's not given default to true" as hacky as it is. :(